src/test/java/io/cdap/plugin/gcp/gcs/sink/GCSBatchSinkCmekKeyTest.java

```diff
@@ -0,0 +1,155 @@
+package io.cdap.plugin.gcp.gcs.sink;
+
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ *  use this file except in compliance with the License. You may obtain a copy of
+ *  the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ *  License for the specific language governing permissions and limitations under
+ *  the License.
+ */
+
+import com.google.common.base.Strings;
+import io.cdap.cdap.etl.api.validation.CauseAttributes;
+import io.cdap.cdap.etl.api.validation.ValidationFailure;
+import io.cdap.cdap.etl.mock.validation.MockFailureCollector;
+import io.cdap.plugin.gcp.common.GCPConfig;
+import io.cdap.plugin.gcp.common.GCPReferenceSinkConfig;
+import org.junit.Assert;
+import org.junit.Assume;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.mockito.internal.util.reflection.FieldSetter;
+
+import java.io.File;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.List;
+import javax.annotation.Nullable;
+
+/**
+ * GCSBatchSink Cmek Key integration test. This test will only be run when below property is provided:
+ * project.id -- the name of the project where staging bucket may be created or new resource needs to be created.
+ * It will default to active google project if you have google cloud client installed.
+ * bucket.path -- the path of the gcs bucket
+ * service.account.file -- the path to the service account key file
+ */
+public class GCSBatchSinkCmekKeyTest {
+  private static String serviceAccountKey;
+  private static String project;
+  private static String serviceAccountFilePath;
+  private static String bucketPath;
+
+  @BeforeClass
+  public static void setupTestClass() throws Exception {
+    // Certain properties need to be configured otherwise the whole tests will be skipped.
+
+    String messageTemplate = "%s is not configured, please refer to javadoc of this class for details.";
+
+    project = System.getProperty("project.id");
+    if (project == null) {
+      project = System.getProperty("GOOGLE_CLOUD_PROJECT");
+    }
+    if (project == null) {
+      project = System.getProperty("GCLOUD_PROJECT");
+    }
+    Assume.assumeFalse(String.format(messageTemplate, "project id"), project == null);
+    System.setProperty("GCLOUD_PROJECT", project);
+
+    bucketPath = System.getProperty("bucket.path");
```

Comment:
this shouldn't be passed in as an argument, as it modifies the expectations of the test.